app/lib/timetable/timetable_edit/lesson/timetable_lesson_edit_page.dart

```diff
@@ -321,31 +338,100 @@ class _EndTimeField extends StatelessWidget {
   }
 }
 
-class _RoomField extends StatelessWidget {
-  const _RoomField(this.initialLesson);
+class RoomField extends StatelessWidget {
+  const RoomField({
+    super.key,
+    required this.onChanged,
+    this.initialPlace,
+  });
 
-  final Lesson initialLesson;
+  final String? initialPlace;
+  final ValueChanged<String> onChanged;
 
   @override
   Widget build(BuildContext context) {
-    final bloc = BlocProvider.of<TimetableEditBloc>(context);
     return Padding(
       padding: const EdgeInsets.only(top: 10),
-      child: Padding(
-        padding: const EdgeInsets.symmetric(horizontal: 16.0),
-        child: PrefilledTextField(
-          prefilledText: initialLesson.place,
+      child: ListTile(
+        leading: const Padding(
+          padding: EdgeInsets.only(left: 6, bottom: 26),
+          child: Icon(Icons.place),
+        ),
+        title: PrefilledTextField(
+          prefilledText: initialPlace,
           decoration: const InputDecoration(
-            icon: Padding(
-              padding: EdgeInsets.only(left: 6),
-              child: Icon(Icons.place),
-            ),
             border: OutlineInputBorder(),
             labelText: "Raum",
           ),
           textCapitalization: TextCapitalization.sentences,
           maxLength: 32,
-          onChanged: bloc.changeRoom,
+          onChanged: onChanged,
+        ),
+      ),
+    );
+  }
+}
+
+class TeacherField extends StatelessWidget {
+  const TeacherField({
+    super.key,
+    required this.teachers,
+    required this.onTeacherChanged,
+    this.initialTeacher,
+  });
+
+  final String? initialTeacher;
+  final CurrentTeachers teachers;
```

Comment:
```suggestion
  final ISet<String> knownTeacherNames;
```

I think using `ISet<String>` is clearer, since otherwise I'd first have to look what `CurrentTeachers` is exactly.